### PR TITLE
Redesign accept invitation screen

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Improve Onborading flow.
Fixes: #PG-2720

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Login/images/01_Create password.png
Comment (bx80) [inline plugins/Login/images/01_Create password.png:1]: Do we know what the license is for this image?

Comment (AltamashShaikh) [reply plugins/Login/images/01_Create password.png:1]: @Javi-Ormaechea Can answer this Q better

Comment (Javi-Ormaechea) [reply plugins/Login/images/01_Create password.png:1]: Thanks for the tag @AltamashShaikh - we own these images @bx80, they were created by our illustrator.

Comment (bx80) [reply plugins/Login/images/01_Create password.png:1]: Thanks for clarifying that @Javi-Ormaechea, I guess these images will then fall under the GPL3 when published and be available for other software to use under the terms of that license.

Comment (AltamashShaikh) [reply plugins/Login/images/01_Create password.png:1]: @bx80 Do we need to specify this anywhere ?

Comment (bx80) [reply plugins/Login/images/01_Create password.png:1]: @AltamashShaikh I don't think so, as far as I'm aware any bundled images will default to the application license. I only mentioned it to confirm that we had the rights to the image and that we're aware that it will become GPL'd :slightly_smiling_face:

File: plugins/Login/lang/en.json
Comment (michalkleiner) [inline plugins/Login/lang/en.json]: 'Confirm' and 'Password' translations already exist in the default lang files (Common). I'm not 100% sure whether we should put the generic 'Confirm password' and 'Create password' there as well and only keep the description key here. What do you reckon @sgiehl?

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [inline plugins/Login/templates/invitation.twig]: Can we gradually start using margins or padding instead of hardcoded line breaks or is there a benefit in `<br>`s over CSS? Especially here when we are also adjusting/adding new CSS. Same for paragraphs with an non-breakable space in them.

File: plugins/Login/images/01_Create password.png
Comment (sgiehl) [reply plugins/Login/images/01_Create password.png:1]: If we want to restrict the usage we can also define our own license for that picture if we want. It for sure needs to allow the usage in Matomo and we need to mention it here: https://github.com/matomo-org/matomo/blob/5.x-dev/LEGALNOTICE

File: plugins/Login/lang/en.json
Comment (sgiehl) [reply plugins/Login/lang/en.json]: Yes. We should reuse existing translation keys where possible. Otherwise it currently doesn't make much difference where new keys are defined. (We had some thoughts around defining everything in the base language file as kind of API in the past, so stuff that wouldn't be changed or removed. But we never really defined that.)

Comment (AltamashShaikh) [reply plugins/Login/lang/en.json]: @michalkleiner @sgiehl Moved to General translation as suggested

File: plugins/Login/templates/invitation.twig
Comment (AltamashShaikh) [reply plugins/Login/templates/invitation.twig]: @michalkleiner done :+1:

File: plugins/Login/images/01_Create password.png
Comment (sgiehl) [inline plugins/Login/images/01_Create password.png:1]: Please give the image a name, that might not cause problems. Using white spaces can always cause problems

File: plugins/Login/templates/invitation.twig
Comment (sgiehl) [inline plugins/Login/templates/invitation.twig]: Personally this doesn't look like a way html and css should be used:
* There is no big sense in having empty `p` tags. This can also be solved using margin/padding. Also I'm not sure if empty paragraphs are maybe even bad in terms of a11y
* Using css classes, which describe what they are doing, like `uppercase` or `m-top` are imho bad. It would almost be the same to directly add a style attribute. You should give html elements classes that make it possible to give them styles using css. And not the other way around. That way you can easily adjust the design without the need of add/removing classes.
* The `create password` actually seems to be the headline of this page. So in terms of a11y this should be a `h1`, which will make it easy to select this element in css even without classes

But that's only my opinion, not sure how @bx80 and @michalkleiner see that.

For me this part would be something like:

```html
<h1>{{ 'General_CreatePassword'|translate }}</h1>
<p class="description">{{ 'Login_CreatePasswordDescription'|translate }}</p>
```

```css

.onboarding {
  h1 {
     text-transform: uppercase;
     line-height: 24px;
     margin-top: 4rem;
  }
  
  p.description {
     line-height: 24px;
     margin: 1em 0;   /* or whatever the empty paragraphs took as space */
  }
}
```

File: plugins/Login/templates/loginLayout.twig
Comment (sgiehl) [inline plugins/Login/templates/loginLayout.twig]: this can be shortened using default
```suggestion
{% set title %}{{ pageTitle|default('Login_LogIn'|translate) }}{% endset %}
```

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: plugins/Login/templates/invitation.twig
Comment (AltamashShaikh) [reply plugins/Login/templates/invitation.twig]: @sgiehl I don't agree that having classes like `uppercase` or `m-top` is bad as frameworks like Bootstrap / Materialize has this kind of classes to make styling easier and reusable.
If you strongly think we should not use it, I am happy to change it :smile: 

I have removed empty p and used margins instead

File: plugins/Login/stylesheets/login.less
Comment (michalkleiner) [inline plugins/Login/stylesheets/login.less]: It's one word, so no capitalisation required in the middle.

```suggestion
.onboarding {
```

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [inline plugins/Login/templates/invitation.twig]:
```suggestion
                        <label for="email">{{ 'UsersManager_Email'|translate }}</label>
```

Comment (AltamashShaikh) [reply plugins/Login/templates/invitation.twig]: @michalkleiner This field is non-editable and adding `for` won't help much as clicking on the label won't focus the input

File: plugins/Login/stylesheets/login.less
Comment (AltamashShaikh) [reply plugins/Login/stylesheets/login.less]: @michalkleiner I think its okay as IDE gives an error of typo in word when we change the camelCase to lowercase for this word

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [reply plugins/Login/templates/invitation.twig]: The markup is not ideal but is better now than what it was first time around.

There are different methodologies around using CSS classes and naming things. For now I think using the boostrap/materialise/tailwind-like approach of utility classes is ok and I thought somewhat consistent with what we already had, being a bit friendlier for developers leaning towards the back-end side of things (as opposed to pure front-end coders). And since HTML can take a lot, we don't focus on a correct semantic and a11y-compliant structure as much as we should.

On the other hand, we might be using a bit more descriptive classnames here as we are already mixing in things like `.footerLinks`, `.formSubmitButton` etc., so adding `.description` with all its styling in one place probably makes sense, too.

I wouldn't stress too much for this change, but we should be aware of the friction and start pushing on unifying the UI componentry and approach to things like this more and more with the product owner and other folks in the wider team so we get consistency, usability and a11y, and not increase technical debt.

File: plugins/Login/stylesheets/login.less
Comment (michalkleiner) [reply plugins/Login/stylesheets/login.less]: The IDE spell check isn't always the smartest one, it needs a bit of help sometimes.
In this case I'd suggest adding the word as a custom word if it finds it incorrect.

https://www.merriam-webster.com/dictionary/onboarding

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [reply plugins/Login/templates/invitation.twig]: It's an a11y aspect. Screen readers can't guess to which form input the label belongs to. 

The label element always either needs to wrap the form element or have the `for` attribute.

File: plugins/Login/templates/invitation.twig
Comment (sgiehl) [reply plugins/Login/templates/invitation.twig]: My intention on having descriptive class names in favor of functional ones is, that in a perfect world - and I know that we are far away from that - we would have a clean HTML structure and descriptive class names for all (relevant) elements.
That would allow additional themes to fully adjust the layout. As themes can only provide CSS and can't overwrite all parts of the HTML easily, classes like `uppercase` won't be helpful anymore. There might afterwards be places where you still want the uppercase but others where you want to remove it, which would then only be possible by using complex rules to overwrite the existing ones. Using element styling and descriptive class names would be a lot more flexible.

Comment (michalkleiner) [reply plugins/Login/templates/invitation.twig]: That's a valid point that I forgot to factor in — theming.

File: plugins/Login/images/01_Create password.png
Comment (AltamashShaikh) [reply plugins/Login/images/01_Create password.png:1]: done

File: plugins/Login/templates/invitation.twig
Comment (AltamashShaikh) [reply plugins/Login/templates/invitation.twig]: @sgiehl Updated as suggested, please check.

File: plugins/Login/stylesheets/login.less
Comment (AltamashShaikh) [reply plugins/Login/stylesheets/login.less]: @michalkleiner updated

File: plugins/Login/templates/invitation.twig
Comment (AltamashShaikh) [reply plugins/Login/templates/invitation.twig]: @michalkleiner updated as suggested

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [inline plugins/Login/templates/invitation.twig:26]: This could be tricky if we have emails that are not valid emails e.g. in local installs... what do you think @sgiehl?

```suggestion
                        <input type="email" name="email" id="email" value="{{ user.email }}" size="20" readonly="readonly"
```

Comment (sgiehl) [reply plugins/Login/templates/invitation.twig:26]: There is actually no bigger benefit in having the type email here. The field is readonly and the value is provided by us. So if the browser validation wold fail and the user can't submit the form, there would be nothing the user could do.

File: plugins/Login/stylesheets/login.less
Comment (sgiehl) [inline plugins/Login/stylesheets/login.less]: Is there any element having this class? Couldn't find one...

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: plugins/Login/stylesheets/login.less
Comment (AltamashShaikh) [reply plugins/Login/stylesheets/login.less]: Removed, I would have added for the additional 2 screens which we moved to CLOUD to now

**Comment by AltamashShaikh** [comment]
@sgiehl @michalkleiner Is this good to merge or do we need more changes ?

File: plugins/Login/templates/invitation.twig
Comment (michalkleiner) [reply plugins/Login/templates/invitation.twig:26]: I missed the readonly attribute, so you are (as always 👍) correct.

File: plugins/Login/templates/loginLayout.twig
Comment (sgiehl) [inline plugins/Login/templates/loginLayout.twig]: That one is actually not needed, as the variable is only used once and could be simplified with a '??` usage
```suggestion
```

File: plugins/Login/templates/loginLayout.twig
Comment (sgiehl) [inline plugins/Login/templates/loginLayout.twig]: see previous comment
```suggestion
            {% include "@CoreHome/_logo.twig" with { 'logoLink': 'https://matomo.org', 'centeredLogo': (isCenteredLogo ?? true), 'useLargeLogo': false } %}
```

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left another comment for a small code improvement. Otherwise this looks good

File: plugins/Login/templates/loginLayout.twig
Comment (AltamashShaikh) [reply plugins/Login/templates/loginLayout.twig]: @sgiehl Updated as suggested

**Comment by sgiehl** [comment]
@AltamashShaikh There is a integration test failing that needs to be fixed.

**Comment by AltamashShaikh** [comment]
> @AltamashShaikh There is a integration test failing that needs to be fixed.

@sgiehl Thanks, should be fixed now

**Comment by sgiehl** [comment]
Ok. Will merge this one, once the tests passed.

**Pull request merged**
